### PR TITLE
UHF-8196: Unpublished content styles

### PR DESCRIPTION
**Pull request by teroelonen**
# [UHF-8196](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8196)

## What was done

* Styles for cards and menu items that are linking to content that is unpublished.
* Classes for templates that should display the unpublished state.

## How to install

* Make sure your instance (sote or kasko are good for example because you need TPR-integration to test this one) is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-8196_unpublished_content_styles`
* Run `make drush-cr`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-8196_unpublished_content_styles`
* Run `make drush-updb drush-cr`
* Testing this feature requires you to add classes manually to navigation items because the logic where the classes are assigned are done in [UHF-8603](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8603) issue.

## How to test

* [x] Create landing page and standard page that is not published. Find TPR-unit and TPR-service that are not published. 
* [x] Now on any content page create content cards paragraph where you link some published and unpublished content.
* [x] Also add service list paragraph and service list search paragraph with some published and unpublished services.
* [x] Go to view the created page as logged in user and you should see the unpublished content with red text colors and borders. Layout references can be checked here: https://share.goabstract.com/15ba1b91-560f-4b6c-b663-d2525f9a30cf?collectionLayerId&mode
* [x] Now go to some TPR-unit and TPR-service pages that list the unpublished items in the blocks on the bottom of the page and they should have the same styling as in the layout reference above.
* [x] With inspector add the class `menu__link--unpublished` to some of the main menu items and see that they are colored red when adding the class. Layout references can be checked here: https://share.goabstract.com/68cf3afb-f167-42c1-9dd0-14dfd19e1cc2?mode=design&sha=9c1ee0a401c7a3af53192f9f2a13b86edf23f5fa
* [x] Check that code follows our standards

## Designers review

* [x] This PR does not need designers review
* [ ] This PR has been visually reviewed by a designer (Name of the designer)

## Other PRs

* https://github.com/City-of-Helsinki/drupal-hdbt/pull/681
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/528

[UHF-8196]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[UHF-8603]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8603?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: hdbt.theme [hunk 1]
```diff
@@ -1317,6 +1317,13 @@ function hdbt_preprocess_tpr_unit(&$variables): void {
   }
   $entity = $variables['entity'];
 
+  // Add unit publish state to variable.
+  $variables['unit_published'] = TRUE;
```
Comment (khalima) [inline hdbt.theme:1321]: This variable could be more generic. Like `$variables['content_entity_published']`

File: hdbt.theme [hunk 2]
```diff
@@ -1346,8 +1353,15 @@ function hdbt_preprocess_tpr_service(&$variables): void {
   if (!isset($variables['entity']) || !$variables['entity'] instanceof Service) {
     return;
   }
+  // Add service publish state to variable.
+  $variables['service_published'] = TRUE;
```
Comment (khalima) [inline hdbt.theme:1357]: This variable could be more generic. Like `$variables['content_entity_published']`

File: templates/module/helfi_tpr/tpr-service.html.twig [hunk 3]
```diff
@@ -37,7 +37,17 @@
 
 {% elseif view_mode == 'teaser' %}
 
-  <div class="service service--teaser">
+  {%
+    set service_classes = [
+      'service',
+      'service--teaser',
+      not service_published ? 'service--teaser--unpublished',
```
Comment (khalima) [inline templates/module/helfi_tpr/tpr-service.html.twig:44]: This variable could be more generic. Like `not content_entity_published ? 'service--teaser--unpublished',`

File: templates/module/helfi_tpr/tpr-unit--minimal.html.twig [hunk 4]
```diff
@@ -14,8 +14,15 @@
     }]) %}
 {% endif %}
 
+{%
+  set card_tpr_classes = [
+  not unit_published ? 'card--unpublished',
```
Comment (khalima) [inline templates/module/helfi_tpr/tpr-unit--minimal.html.twig:19]: This variable could be more generic. Like `not content_entity_published ? 'card--unpublished',`

File: templates/module/helfi_tpr/tpr-unit--teaser-with-image.html.twig [hunk 5]
```diff
@@ -12,8 +12,15 @@
   {% endif %}
 {% endset %}
 
+{%
+  set card_tpr_classes = [
+    not unit_published ? 'card--unpublished',
```
Comment (khalima) [inline templates/module/helfi_tpr/tpr-unit--teaser-with-image.html.twig:17]: This variable could be more generic. Like `not content_entity_published ? 'card--unpublished',`

File: templates/module/helfi_tpr/tpr-unit--wide-teaser.html.twig [hunk 6]
```diff
@@ -1,17 +1,25 @@
+{%
+  set content_liftup_classes = [
+    'content-liftup',
+    not unit_published ? 'content-liftup--unpublished',
```
Comment (khalima) [inline templates/module/helfi_tpr/tpr-unit--wide-teaser.html.twig:4]: This variable could be more generic. Like `not content_entity_published ? 'content-liftup--unpublished',`

File: templates/module/helfi_tpr/tpr-service--teaser-search-result.html.twig [hunk 7]
```diff
@@ -1,5 +1,12 @@
+{%
+  set card_tpr_classes = [
+  not service_published ? 'card--unpublished',
```
Comment (khalima) [inline templates/module/helfi_tpr/tpr-service--teaser-search-result.html.twig:3]: This variable could be more generic. Like `not content_entity_published ? 'card--unpublished',`

**Review by khalima — CHANGES_REQUESTED** [review]
The `unit_published` and `service_published` variables could be named in identically

File: hdbt.theme (hunk 1 above)
Comment (teroelonen) [reply hdbt.theme]: Made it generic 🦖

File: templates/module/helfi_tpr/tpr-service.html.twig (hunk 3 above)
Comment (teroelonen) [reply templates/module/helfi_tpr/tpr-service.html.twig]: Made it generic 🦖

File: templates/module/helfi_tpr/tpr-unit--minimal.html.twig (hunk 4 above)
Comment (teroelonen) [reply templates/module/helfi_tpr/tpr-unit--minimal.html.twig]: Made it generic 🦖

File: templates/module/helfi_tpr/tpr-unit--teaser-with-image.html.twig (hunk 5 above)
Comment (teroelonen) [reply templates/module/helfi_tpr/tpr-unit--teaser-with-image.html.twig]: Made it generic 🦖

File: templates/module/helfi_tpr/tpr-unit--wide-teaser.html.twig (hunk 6 above)
Comment (teroelonen) [reply templates/module/helfi_tpr/tpr-unit--wide-teaser.html.twig]: Made it generic 🦖

File: templates/module/helfi_tpr/tpr-service--teaser-search-result.html.twig (hunk 7 above)
Comment (teroelonen) [reply templates/module/helfi_tpr/tpr-service--teaser-search-result.html.twig]: Made it generic 🦖

File: hdbt.theme (hunk 2 above)
Comment (teroelonen) [reply hdbt.theme]: Made it generic 🦖

**Review by khalima — APPROVED** [review]
Excellent. 🚀

**Pull request merged**